src/main/java/net/fabricmc/loader/impl/launch/server/FabricServerLauncher.java

```diff
@@ -64,8 +66,8 @@ public static void main(String[] args) {
 
 		try {
 			Class<?> c = Class.forName(mainClass);
-			c.getMethod("main", String[].class).invoke(null, (Object) args);
-		} catch (Exception e) {
+			MethodHandles.lookup().findStatic(c, "main", MethodType.methodType(Void.TYPE, String[].class)).invokeExact(args);
```

Comment:
it is not the collection toArray (Object[]), but one that returns String[], so this should be fine

Reply:
This is not client/vs server launcher, this (in most if not all cases) calls Knot (see line 38), and then Knot calls the game provider too. I think the state would be very messed up if this can't find Knot, and I don't think it's worth catching separately (in the other case I kept the split because it was already there).

Reply:
Then it's fine.